text/3857-cfg-version.md

```diff
@@ -0,0 +1,635 @@
+- Treat the nightly as not implementing that language version at all, only the previous
+- Leave a marker that that language version is incomplete, while the previous language version is complete
+
+In RFC 2523, this was left as an
+[unresolved question](https://rust-lang.github.io/rfcs/2523-cfg-path-version.html#unresolved-questions).
+
+The initial implementation treated nightlies as complete.
+This was [changed to incomplete](https://github.com/rust-lang/rust/pull/72001) after
+[some discussion](https://github.com/rust-lang/rust/issues/64796#issuecomment-624673206).
+In particular, this is important for
+- the case of package `bleeding-edge` starting to use a new feature behind `#[cfg(since)]` and package `nightly-only` has their toolchain pinned to a nightly before the feature was stabilized (to ensure consistent behavior of unstable features), package `nightly-only` cannot add or update their dependency on `bleeding-edge` without getting a "feature gate needed" error.
+- bisecting nightlies.
+
+This was [changed back to complete](https://github.com/rust-lang/rust/pull/81468) after
+[some more discussion](https://github.com/rust-lang/rust/issues/64796#issuecomment-634546711).
+In particular, this is important for
+- keeping friction down for packages preparing for stabilized-on-nightly features as their `#[cfg(since)]`s can be inserted and "just work" which can be important for getting feedback quickly while the feature is easier to adapt to feedback that can be gained from these users
+  - releasing the package while its in this state puts it at risk to be broken if the feature is changed after stabilization
+
+For RFC 2523, they settled on pre-releases being incomplete,
+favoring maintainers to adopt stabilized-on-nightly features immediately
+while letting people on pinned nightlies or bisecting nightlies to set a `-Z` to mark the version as incomplete.
+
+In this RFC, we settled on translating `-nightly` to `-incomplete` because:
+- Maintainers can adopt stabilized-on-nightly features with `#[cfg(since(rust, "1.100.0-0"))]` (the lowest pre-release for `1.100.0`), keeping friction low while explicitly acknowledging that the unstable feature may change
+- Allows build scripts to experiment with other logic when approximating the vendor version from the language version with less of a chance of needing to invoke `rustc` (e.g. detecting nightly)
+- It provides extra context when approximating the vendor version from the language version when populating build information
+
+We called the pre-release `-incomplete` to speak to the relationship to the language version.
+Other terms like `partial` could as easily apply.
+The term `-nightly` would be more discoverable but would convey more of a relationship to the vendor than the language.
+
+As for differentiating between nightlies,
+that corresponds more to the vendor version than the language version,
+so we do not include that information.
+
+## Alternative designs
```

Comment:
Could we have a vote on the proposed "since" versus the alternative designs?

I personally like them better.

Reply:
The "vote" will be done by the owning team for this RFC (T-lang).  The responsibility for the rest of us is to provide meaningful feedback that can be included for them to review in their evaluation.

Keep in mind though that part of why this RFC came about is concerns that came up during the *stabilization* of `rust_version(_)` which makes it unlikely we will pivot in that direction.